### PR TITLE
Support getMessage method for caught exception

### LINKED ISSUE #662: Support getMessage method on Exceptions
**Summary or problem description**

When you catch an exception in a smart contract and want to get the message string with the `getMessage` method, the contract will fail at runtime. On the NeoVM the java `Exception` object is just the string that has been passed to the `Exception` constructor (or "error" when no string was provided). Thus, when calling `getMessage` on that string, the NeoVM fails.

**Do you have any solution you want to propose?**
Adapt the neow3j compiler such that calls to `Exception.getMessage()` are handled specially. In theory the call to `getMessage()` simply has to be ignored since the exception string is already at the top of the stack.

Tasks:
- [x] Adapt the compiler
- [x] Write integration tests in which exceptions are thrown (e.g., by assertions) and then the `Exception` object is handled in the `catch` clause by calling its `getMessage()` method.
- [x] Update the section Smart Contract Development -> NeowJava -> exceptions in the neow3j-docs.

### PR DESCRIPTION
**Pull request by mialbu**
Resolves #662

### CODE REVIEW

**Comment by gsmachado** [comment]
why the int tests are failing?

It seems related to some kind of problem with cache, that @csmuller changed recently.

I will trigger it again just to see if the problem vanishes.

**Comment by mialbu** [comment]
Yes, it's some cache issue in the `Set up JDK 1.8` task. @csmuller is looking into it.

**Comment by mialbu** [comment]
Btw, the tests passed in the commit before and the latest commit only contains code style fixes.

File: compiler/src/main/java/io/neow3j/compiler/converters/MethodsConverter.java [hunk 1]
```diff
@@ -194,6 +197,12 @@ private static AbstractInsnNode handleUncachedMethodCall(NeoMethod callingNeoMet
         return methodInsn;
     }
 
+    private static boolean isThrowableGetMessage(ClassNode owner, MethodInsnNode methodInsn) {
+        return getFullyQualifiedNameForInternalName(owner.name).equals(
+                Throwable.class.getCanonicalName()) &&
+                methodInsn.name.equals(THROWABLE_GET_MESSAGE);
```
Comment (csmuller) [inline compiler/src/main/java/io/neow3j/compiler/converters/MethodsConverter.java:203]: Wait, so the owner is `Throwable` and not `Exception` or `AssertionError`? That's convenient.

**Review by csmuller — APPROVED** [review]
👍

File: compiler/src/main/java/io/neow3j/compiler/converters/MethodsConverter.java (hunk 1 above)
Comment (mialbu) [reply compiler/src/main/java/io/neow3j/compiler/converters/MethodsConverter.java:203]: Yes, both `AssertionError` and `Exception` use the method `getMessage` from their shared super class `Throwable`, which is super convenient. 🙂

**Pull request merged**